### PR TITLE
added upgrade hook for control plane upgrade

### PR DESCRIPTION
**Pull request by gdsoumya**
Signed-off-by: Soumya Ghosh Dastidar <gdsoumya@gmail.com>

#### What this PR does / why we need it:

This PR adds upgrade hook for control plane upgrade agent

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] DCO signed
- [x] Chart Version bumped
- [ ] Variables are documented in the README.md

### CODE REVIEW

File: charts/litmus/values.yaml [hunk 1]
```diff
@@ -38,6 +38,13 @@ ingress:
   #  - secretName: chart-example-tls
   #    hosts: []
 
+upgrade-agent:
+  control-plane:
```
Comment (ispeakc0de) [inline charts/litmus/values.yaml:42]: You can't use hypne here, use camel-case instead

**Review by ispeakc0de — APPROVED** [review]

File: charts/litmus/values.yaml [hunk 2]
```diff
@@ -38,6 +38,13 @@ ingress:
   #  - secretName: chart-example-tls
   #    hosts: []
 
+upgradeAgent:
+  controlPlane:
+    image:
+      repository: upgrade-agent-cp
```
Comment (Jasstkn) [inline charts/litmus/values.yaml:44]: I think that you need to specify repository, like `litmuschaos/upgrade-agent-cp`

**Review by Jasstkn — CHANGES_REQUESTED** [review]

File: charts/litmus/values.yaml (hunk 2 above)
Comment (gdsoumya) [reply charts/litmus/values.yaml:44]: I think this is correct we have a separate global `imageRegistryName` value

Comment (gdsoumya) [reply charts/litmus/values.yaml:44]: this is the complete value for the container image `{{ .Values.image.imageRegistryName }}/{{ .Values.upgradeAgent.controlPlane.image.repository }}:{{ .Values.upgradeAgent.controlPlane.image.tag }}`

**Review by Jasstkn — APPROVED** [review]

**Review by Jasstkn — COMMENTED** [review]
@gdsoumya better to run helm-docs unless it will disturb every PR next times

**Comment by gdsoumya** [comment]
> @gdsoumya better to run helm-docs unless it will disturb every PR next times

Have run that but for some reason it's still giving that error, discussed this issue with @ispeakc0de will figure it out later

**Comment by Jasstkn** [comment]
> > @gdsoumya better to run helm-docs unless it will disturb every PR next times
> 
> Have run that but for some reason it's still giving that error, discussed this issue with @ispeakc0de will figure it out later

you've delete these lines https://github.com/litmuschaos/litmus-helm/pull/155/files#diff-92ea66ab50f5b36835005f51ae021b4830451ef1cedba649b0c19ca727135ca7L143

**Comment by gdsoumya** [comment]
> you've delete these lines https://github.com/litmuschaos/litmus-helm/pull/155/files#diff-92ea66ab50f5b36835005f51ae021b4830451ef1cedba649b0c19ca727135ca7L143

That's the thing I did not delete anything in the readme manually, it was all auto generated

**Comment by gdsoumya** [comment]
I'll manually add it and see if that fixes the error

**Comment by Jasstkn** [comment]
> I'll manually add it and see if that fixes the error

works properly for me. you forgot new line at the end.

**Pull request merged**